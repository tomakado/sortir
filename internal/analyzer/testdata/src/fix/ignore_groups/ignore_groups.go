package ignore_groups

const (
	Group1Z = 3
	Group1A = 1 // want "variable/constant declarations are not sorted"
	Group1B = 2

	Group2Z = "z"
	Group2A = "a"
	Group2B = "b"
)

var (
	z1 = "z1"
	a1 = "a1" // want "variable/constant declarations are not sorted"
	b1 = "b1"

	z2 = "z2"
	a2 = "a2"
	b2 = "b2"
)

type MyStruct struct {
	Z1 int
	A1 int // want "struct fields are not sorted"
	B1 int

	Z2 string
	A2 string
	B2 string
}

type MyInterface interface {
	ZMethod()
	AMethod() // want "interface methods are not sorted"
	BMethod()

	ZMethod2()
	AMethod2()
	BMethod2()
}