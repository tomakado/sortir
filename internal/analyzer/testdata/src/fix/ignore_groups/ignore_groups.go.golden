package ignore_groups

const (
	Group1A = 1 // want "variable/constant declarations are not sorted"
	Group1B = 2
	Group1Z = 3
	Group2A = "a"
	Group2B = "b"
	Group2Z = "z"
)

var (
	a1 = "a1" // want "variable/constant declarations are not sorted"
	a2 = "a2"
	b1 = "b1"
	b2 = "b2"
	z1 = "z1"
	z2 = "z2"
)

type MyStruct struct {
	A1 int // want "struct fields are not sorted"
	A2 string
	B1 int
	B2 string
	Z1 int
	Z2 string
}

type MyInterface interface {
	AMethod() // want "interface methods are not sorted"
	AMethod2()
	BMethod()
	BMethod2()
	ZMethod()
	ZMethod2()
}